### ISSUE TITLE
Differentiate between pre-existing nodes, and not pre-existing nodes

### ISSUE THREAD

**Issue by jmeas** [marionettejs/backbone.marionette#2154]
There are two general categories of views in Backbone. One category are pre-existing nodes that are unlikely to be rendered ever. An example of this would be creating a `rootView` and passing the `body` as the element, in a situation where the `body` already has a handful of nodes. The other category are views which may or may not have templates and are rendered throughout the application's lifecycle. The latter of the two categories is by far the most common.

Marionette does not distinguish between these two types of views, which creates some bad side effects. For instance, both sorts of LayoutViews are treated identically, which leads to [some nasty bugs](https://github.com/marionettejs/backbone.marionette/issues/1971).

One way to resolve situations like that are to differentiate between these two use-cases. We can infer that a view exists based on two criteria: it has no template, and it has children nodes.

In other words:

```js
_preExistingNode: function() {
  return !this.template && this.$el.children().length;
}
```

During the instantiation phase of any view, this would be used to dictate the behavior of the view. Take a LayoutView, for instance. Pre-existing nodes would have their regions instantiated instantly, and their UI would also be bound instantly. LayoutViews that aren't pre-existing would wait until the first render before doing these actions.

**Comment by paulfalgout** [comment]
Pretty sure this is resolved in v3.1
